core/arch/x86/decode_table.c

```diff
@@ -1406,6 +1462,28 @@ const instr_info_t * const op_instr[] =
 #define Lss TYPE_L, OPSZ_4_of_16 /* immed is 1 byte but reg is xmm/ymm */
 #define Lsd TYPE_L, OPSZ_8_of_16 /* immed is 1 byte but reg is xmm/ymm */
 
+/* AVX-512 additions */
+#define KRb TYPE_K_REG, OPSZ_1
+#define KRw TYPE_K_REG, OPSZ_2
+#define KRd TYPE_K_REG, OPSZ_4
+#define KRq TYPE_K_REG, OPSZ_8
+#define KRMb TYPE_K_MODRM_R, OPSZ_1
+#define KRMw TYPE_K_MODRM_R, OPSZ_2
+#define KRMd TYPE_K_MODRM_R, OPSZ_4
+#define KRMq TYPE_K_MODRM_R, OPSZ_8
+#define KMRMb TYPE_K_MODRM, OPSZ_1
```

Comment:
How about modeling after types P, Q, and R?  So s/KR/KP/; s/KRM/KR/; s/KMRM/KQ/.  It's then closer to existing names and the table stays compact.  WDYT?

Reply:
Ok, fine with that. Done.